src/passes/Asyncify.cpp

```diff
@@ -380,8 +381,8 @@ class FakeGlobalHelper {
   }
 
 private:
-  std::map<Type, Name> map;
-  std::map<Name, Type> rev;
+  std::unordered_map<Type, Name> map;
+  std::unordered_map<Name, Type> rev;
```

Comment:
How about for changes like this, we create a new `uniterable_map` (or better name), which is `unordered_map` without the ability to iterate? That would avoid the possibility of nondeterminism, which otherwise can be very annoying to find later if we miss anything in this PR.

In other words, the PR could change a `map` into either `uniterable_map` (if we never iterate on it) or `ordered_map` (if we do). Then we should have 0 risk.

Reply:
If we had an `uniterable_map` available in a header, I have a hard time imagining that I would use it beyond this single PR. The tough part about getting the determinism correct here is that now getting deterministic iteration depends on having deterministic insertions, whereas previously the insertions could be non-deterministic and the iteration would still be deterministic due to sorting. An `uniterable_map` would not help solve that problem, unfortunately.

Reply:
I see, very good point...

Yes, maybe not that generally useful then. Still I think it's worthwhile, but can discuss separately if it doesn't help you here.